### DISCUSSION TITLE
Can I save the GraphicsView's Canvas as a Bitmap?

### DISCUSSION THREAD

**Discussion by chinadyz** [dotnet/maui#8969] [Q&A] [answered]
In Skiasharp, we can create a Bitmap in memory,  then draw it in a canvas, and then display the image to the interface.

In Winform, we can new a Graphics in memory, and then set it as a picturebox's image, for example:

```
            Bitmap BmpTemp = new Bitmap(Panel_Width, Panel_Height);
            Graphics MyGraphics = Graphics.FromImage(BmpTemp);
            MyGraphics.DrawImage(BmpTemp, 0, 0, Panel_Width, Panel_Height);
            ……
            this.PictureBox1.Image = BmpTemp;
```
But in GraphicsView's documentation I didn't find a corresponding description.

The function I want to achieve is:
**Draw in the Canvas of GraphicsView, then save it to memory as a Bitmap, and then display it directly in an image control of maui (similar to the Winform sample code above).**

In Maui's documentation it says "Images can be saved by the and methods. Each method saves the to a , and enables optional and quality values to be specified.IImage.SaveIImage.SaveAsyncIImageStreamImageFormat", but the code in it only involves the image's Zoom directly, without being associated with Canvas.

Excuse me, is there any way I can achieve the desired function?

**Comment by xtuzy** [comment, marked as answer]
You can do it by youselft, canvas have a platformcanvas, you can search every platform how to use this platformcanvas draw on bitmap.  I realize offscreen draw have some related code  https://github.com/xtuzy/MauiPlayground/tree/master/MauiLib/GraphicExtension

**Reply by chinadyz** [reply]
I just studied your code, It is useful！
Thanks.
